test/e2e/nns_update_timestamp_test.go

```diff
@@ -0,0 +1,47 @@
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var _ = Describe("NNS update timestamp", func() {
+	Context("when updating nns", func() {
+		It("timestamp should be changed", func() {
+			for _, node := range nodes {
+				var interval int = 15
+				key := types.NamespacedName{Name: node}
+				timeBefore := nodeNetworkState(key).Status.LastSuccessfulUpdate
+
+				if kubeconfig := os.Getenv("KUBECONFIG"); len(kubeconfig) > 0 {
+					config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+					ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+					clientset, err := kubernetes.NewForConfig(config)
+					cs, err := clientset.CoreV1().ConfigMaps("nmstate").Get("nmstate-config", metav1.GetOptions{})
+					ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+					interval, err = strconv.Atoi(cs.Data["node_network_state_refresh_interval"])
+					ExpectWithOffset(1, err).ToNot(HaveOccurred())
+					interval = interval * 3
+				}
+
+				time.Sleep(time.Duration(interval) * time.Second)
+				timeAtfer := nodeNetworkState(key).Status.LastSuccessfulUpdate
+				Expect(timeAtfer.After(timeBefore.Time)).To(BeTrue())
```

Comment:
Use gomega matchers so you don't need the fmt.Printf the matcher will show you the info that does not match
```go
Expect(timeAfter).To(BeTemporally(">", timeBefore))
```

Reply:
hope what i did now is good , (used Eventually and found the phrase that has boolean value, no need assert since the assert is in the Eventually)